app/src/main/java/chat/rocket/android/service/ConnectivityManagerImpl.java

```diff
@@ -0,0 +1,230 @@
+package chat.rocket.android.service;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import chat.rocket.android.helper.RxHelper;
+import chat.rocket.android.log.RCLog;
+import hugo.weaving.DebugLog;
+import rx.Observable;
+import rx.Single;
+import rx.subjects.PublishSubject;
+
+/**
+ * Connectivity management implementation.
+ */
+/*package*/ class ConnectivityManagerImpl implements ConnectivityManagerApi, ConnectivityManagerInternal {
```

Comment:
This `ConnectivityManagerImpl` talks directly with `ServerInfoImpl` (which is really a `RealmServerInfo`).

So maybe `RealmConnectivityManager` would be a better name.